### PR TITLE
Fix formatting for init followed by a paren

### PR DESCRIPTION
**Pull request by kimdv**

### CODE REVIEW

File: Sources/SwiftBasicFormat/generated/BasicFormat.swift [hunk 1]
```diff
@@ -174,12 +174,13 @@ open class BasicFormat: SyntaxRewriter {
...
-    case (.keyword(.as), .exclamationMark), // Ensures there is not space in `as!`
-     (.keyword(.as), .postfixQuestionMark), // Ensures there is not space in `as?`
-     (.exclamationMark, .leftParen), // Ensures there is not space in `myOptionalClosure!()`
+    case (.exclamationMark, .leftParen), // Ensures there is not space in `myOptionalClosure!()`
      (.exclamationMark, .period), // Ensures there is not space in `myOptionalBar!.foo()`
+     (.keyword(.as), .exclamationMark), // Ensures there is not space in `as!`
+     (.keyword(.as), .postfixQuestionMark), // Ensures there is not space in `as?`
      (.keyword(.`init`), .leftParen), // Ensures there is not space in `init()`
      (.keyword(.`init`), .postfixQuestionMark), // Ensures there is not space in `init?`
+     (.keyword(.`init`), .rightParen), // Ensures there is not space in `.map(Foo.init)`
```
Comment (kimdv) [inline Sources/SwiftBasicFormat/generated/BasicFormat.swift:183]: Maybe we should make it more general? So when I have a something `["foo", "bar", "baz"].map(MyStruct.someStaticMethod)`

Comment (ahoppen) [reply Sources/SwiftBasicFormat/generated/BasicFormat.swift:183]: I just merged https://github.com/apple/swift-syntax/pull/1298, which should allow you to override `requires_trailing_space` for `Name` in a `MemberAccessExpr`. 

Which just got me thinking whether `init` shouldn’t require a trailing space by default. I can’t think of a place where it does require a trailing space right now.

**Comment by kimdv** [comment]
@swift-ci please test

File: Tests/SwiftSyntaxBuilderTest/ReturnStmsTests.swift [hunk 2]
```diff
@@ -45,6 +37,10 @@ final class ReturnStmtTests: XCTestCase {
         StmtSyntax("return !myBool"),
         "return !myBool"
       ),
+      #line: (
+        StmtSyntax("return data.child(at: 2, parent: Syntax(self)).map(UnexpectedNodesSyntax.init)"),
+        "return data.child(at: 2, parent: Syntax(self)).map(UnexpectedNodesSyntax.init)"
```
Comment (kimdv) [inline Tests/SwiftSyntaxBuilderTest/ReturnStmsTests.swift:42]: Added a test case to ensure don't add a leading/trailing space in this example

**Review by ahoppen — APPROVED** [review]
Nice

**Pull request merged**